### PR TITLE
chore!: Rewrite Lint Workflow to Github

### PR DESCRIPTION
**Pull request by oliverbaehler**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by oliverbaehler** [comment]
Before this PR can be merged, the following two PRs must be merged (since the block the linting and would fail on every future release):

  * https://github.com/argoproj/argo-helm/pull/688
  * https://github.com/argoproj/argo-helm/pull/687

**Comment by oliverbaehler** [comment]
This PR implements just the linting part of https://github.com/argoproj/argo-helm/pull/652. @alexec please take a look if this makes you feel more comfortable. If we can merge this after the required prs are also merged. And after that we can look into rewriting the publishing component.

File: .github/workflows/lint-and-test.yml
Comment (mkilchhofer) [inline .github/workflows/lint-and-test.yml:37]: Can we move both the `ct.yaml` and the `lintconf.yaml` to the `.github` directory? I don't like having these files in the root directory :-)

File: charts/argo-events/ci/test-values.yaml
Comment (mkilchhofer) [inline charts/argo-events/ci/test-values.yaml:4]: Whats the purpose of this change? Does the new workflow fail?

**Review by mkilchhofer — COMMENTED** [review]
Found 2 things to discuss, please have a look into my comments.

File: charts/argo-events/ci/test-values.yaml
Comment (mkilchhofer) [reply charts/argo-events/ci/test-values.yaml:4]: Ahh, I see. The current workflow does not install the chart (eg. on a KinD cluster). And now with the new chart-testing GitHub action, we perform linting AND install tests.

**Review by mkilchhofer — APPROVED** [review]

**Comment by oliverbaehler** [comment]
@alexec @alexmt We currently can't merge this pr. The problem being that the `ci/circleci:lint` is required before merge. We can't change this settings, since we are not repository owners. Could one of you remove the requirment for the `ci/circleci:lint` job and add the requirment on the `Linting and Testing`job, which newly lints and installs all the charts.

**Comment by oliverbaehler** [comment]
Removed Argo Chart since it was deprecated with #668

**Comment by oliverbaehler** [comment]
@mkilchhofer says is not good idea :thinking:

**Review by mkilchhofer — APPROVED** [review]
LGTM

**Pull request merged**
